core/workspace_comment_svg.js

```diff
@@ -0,0 +1,591 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Object representing a code comment on a rendered workspace.
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+goog.provide('Blockly.WorkspaceCommentSvg');
+
+goog.require('Blockly.Events.CommentCreate');
+goog.require('Blockly.Events.CommentDelete');
+goog.require('Blockly.Events.CommentMove');
+goog.require('Blockly.WorkspaceComment');
+
+
+/**
+ * Class for a workspace comment's SVG representation.
+ * @param {!Blockly.Workspace} workspace The block's workspace.
+ * @param {string} content The content of this workspace comment.
+ * @param {number} height Height of the comment.
+ * @param {number} width Width of the comment.
+ * @param {string=} opt_id Optional ID.  Use this ID if provided, otherwise
+ *     create a new ID.
+ * @extends {Blockly.WorkspaceComment}
+ * @constructor
+ */
+Blockly.WorkspaceCommentSvg = function(workspace, content, height, width,
+    opt_id) {
+  // Create core elements for the block.
+  /**
+   * @type {SVGElement}
+   * @private
+   */
+  this.svgGroup_ = Blockly.utils.createSvgElement(
+      'g', {'class': 'blocklyComment'}, null);
+  this.svgGroup_.translate_ = '';
+
+  this.svgRect_ = Blockly.utils.createSvgElement(
+      'rect',
+      {
+        'class': 'blocklyCommentRect',
+        'x': 0,
+        'y': 0,
+        'rx': Blockly.WorkspaceCommentSvg.BORDER_RADIUS,
+        'ry': Blockly.WorkspaceCommentSvg.BORDER_RADIUS
+      });
+  this.svgGroup_.appendChild(this.svgRect_);
+
+  /**
+   * Whether the comment is rendered onscreen and is a part of the DOM.
+   * @type {boolean}
+   * @private
+   */
+  this.rendered_ = false;
+
+  /**
+   * Whether to move the comment to the drag surface when it is dragged.
+   * True if it should move, false if it should be translated directly.
+   * @type {boolean}
+   * @private
+   */
+  this.useDragSurface_ =
+      Blockly.utils.is3dSupported() && !!workspace.blockDragSurface_;
+
+  Blockly.WorkspaceCommentSvg.superClass_.constructor.call(this,
+      workspace, content, height, width, opt_id);
+
+  this.render();
+}; goog.inherits(Blockly.WorkspaceCommentSvg, Blockly.WorkspaceComment);
+
+/**
+ * The width and height to use to size a workspace comment when it is first
+ * added, before it has been edited by the user.
+ * @type {number}
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.DEFAULT_SIZE = 100;
+
+/**
+ * Dispose of this comment.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.dispose = function() {
+  if (!this.workspace) {
+    // The comment has already been deleted.
+    return;
+  }
+  // If this comment is being dragged, unlink the mouse events.
```

Comment:
should this say "deleted" instead of "dragged"?

Reply:
Done.